libkbfs/folder_block_manager.go

```diff
@@ -404,40 +425,46 @@ func (fbm *folderBlockManager) processBlocksToDelete(ctx context.Context, toDele
 	// part of the folder history.  (This could happen
 	// if the Sync was canceled while the MD put was
 	// outstanding.)
-	rmds, err := getMDRange(ctx, fbm.config, fbm.id, toDelete.md.BID,
-		toDelete.md.Revision, toDelete.md.Revision, toDelete.md.MergedStatus())
-	if err != nil || len(rmds) == 0 {
-		// Don't re-enqueue immediately, since this might mean no new
-		// revision has made it to the server yet, and we'd just get
-		// into an infinite loop.
-		fbm.enqueueBlocksToDeleteAfterShortDelay(toDelete)
-		return nil
-	}
-	dirsEqual, err := CodecEqual(fbm.config.Codec(),
-		rmds[0].data.Dir, toDelete.md.data.Dir)
-	if err != nil {
-		fbm.log.CErrorf(ctx, "Error when comparing dirs: %v", err)
-	} else if dirsEqual {
-		// This md is part of the history of the folder,
-		// so we shouldn't delete the blocks.
-		fbm.log.CDebugf(ctx, "Not deleting blocks from revision %d",
+	if toDelete.bdType == blockDeleteOnMDFail {
+		rmds, err := getMDRange(ctx, fbm.config, fbm.id, toDelete.md.BID,
+			toDelete.md.Revision, toDelete.md.Revision,
+			toDelete.md.MergedStatus())
+		if err != nil || len(rmds) == 0 {
+			// Don't re-enqueue immediately, since this might mean no new
+			// revision has made it to the server yet, and we'd just get
+			// into an infinite loop.
+			fbm.enqueueBlocksToDeleteAfterShortDelay(toDelete)
+			return nil
+		}
+		dirsEqual, err := CodecEqual(fbm.config.Codec(),
+			rmds[0].data.Dir, toDelete.md.data.Dir)
+		if err != nil {
+			fbm.log.CErrorf(ctx, "Error when comparing dirs: %v", err)
+		} else if dirsEqual {
+			// This md is part of the history of the folder,
+			// so we shouldn't delete the blocks.
+			fbm.log.CDebugf(ctx, "Not deleting blocks from revision %d",
+				toDelete.md.Revision)
```

Comment:
any reason you switch from `toDelete.md` to `rmds[0]`? They should be the same, right?